design/GraphClientsDesign.md

```diff
@@ -0,0 +1,52 @@
+
+### Graph JS Deliverables:
+
+Note: Package names yet to be decided
+
+1. `microsoftgraph/microsoft-graph-javascript-service`: 
```

Comment:
this is a clearer name for sure, but what is the reason for breaking the pattern: `msgraph-sdk-<language>`?

Reply:
I'm still debating this to be honest. @nikithauc I think we should provide unicity with the rest of the SDKs. For now I'd love to think @microsoft/msgraph-sdk-javascript. We never name "service" in any SDK. We do for the core, though.

Reply:
When I saw `microsoftgraph/`, I thought these were referring to the GitHub repositories.

I think the package naming doesn't follow the pattern I mentioned. These are the current published packages:
https://www.npmjs.com/~microsoftgraph

---

**msgraph-sdk-javascript**
Microsoft Graph Client Library
published 0.3.1 • *5 years ago*

**microsoft-graph-typings**
Types for Microsoft Graph objects
published 0.3.0 • *5 years ago*

**@microsoft/microsoft-graph-types**
Types for Microsoft Graph objects
published 2.11.0 • *a month ago*

**@microsoft/microsoft-graph-client**
Microsoft Graph Client Library
published 3.0.1 • *2 months ago*

**@microsoft/microsoft-graph-types-beta**
Beta version of Typings for Microsoft Graph objects
published 0.21.0-preview • *5 days ago*

Reply:
The package names are just examples and the use of the service and core prefix is just for clarity purposes. 
These are not the final names.